pypeit/spectrographs/keck_mosfire.py

```diff
@@ -347,7 +372,214 @@ def configuration_keys(self):
             and used to constuct the :class:`~pypeit.metadata.PypeItMetaData`
             object.
         """
-        return ['decker', 'dispname', 'filter1']
+        return ['decker_basename', 'slitlength', 'slitwid', 'dispname', 'filter1']
+
+    def modify_config(self, fitstbl, cfg):
+        """
+        Modify the configuration dictionary for a given frame. This method is used
+        in :func:`pypeit.metadata.PypeItMetaData.set_configurations` to modify in place
+        the configuration requirement to assign a specific frame to the current setup.
+
+        Args:
+            fitstbl(`astropy.table.Table`_):
+                The table with the metadata for one frames.
+            cfg (:obj:`dict`):
+                dictionary with metadata associated to a specific configuration.
+
+        Returns:
+            :obj:`dict`: modified dictionary with metadata associated to a specific configuration.
+        """
+        if fitstbl['decker'] is not None and cfg['decker_basename'] is not None and 'LONGSLIT' in fitstbl['decker'] \
+                and 'LONGSLIT' in cfg['decker_basename'] and 'science' not in fitstbl['frametype'] and\
+                'standard' not in fitstbl['frametype'] and fitstbl['slitlength'] == 46.:
+            cfg2 = copy.deepcopy(cfg)
+            cfg2.pop('slitlength')
+            return cfg2
+        if fitstbl['decker'] is not None and cfg['decker_basename'] is not None and 'long2pos' in fitstbl['decker'] and \
+                'long2pos' in cfg['decker_basename'] and 'science' not in fitstbl['frametype'] \
+                and 'standard' not in fitstbl['frametype']:
+            cfg2 = copy.deepcopy(cfg)
+            cfg2['decker_basename'] = 'long2pos'
+            return cfg2
+        return cfg
+
+    def get_comb_group(self, fitstbl):
+        """
+
+        This method modify comb_id and bkg_id metas for a specific instrument.
```

Comment:
Docs are too terse. Explain that this routine parses the dither pattern etc. to set the ids. It is doing more than just modifying ids.

Reply:
Also, please look at the parse dither pattern method that is also defined for mosfire, and figure out if you can combine that codes functionality with what you are doing here, i.e. it appears you are also parsing dither patterns from header cards to set comb and bg_ids. We do not want multiple pieces of code that do similar things. Perhaps we can get rid of parse_dither_pattern.

Reply:
Added more docs.

Reply:
I don't think it's possible at these stage to combine this method with `parse_dither_pattern`, without additional substantial changes. 
`parse_dither_pattern` is a little more generic in the sense that takes as input a list of files, while `get_comb_group` uses the fitstbl for the selected calibration group. 
As it is now, `parse_dither_pattern` is useful to be used also outside of the main reduction. Moreover, `parse_dither_pattern` is also defined for `vlt_fors`.

Reply:
Hi @deborape I'm not satisfied with your response here. The docs still don't state that this routine sets the comb and bg_ids by parsing the dither pattern.  That needs to be added to the docs. Also, I'm not convinced by your argument that this routine and parse_dither_pattern cannot be merged. It is a flaw that parse_dither_pattern is written to take a list of files which is admittedly my fault.  You could easily compute the fitstable where the files are being used. I don't like that we have two routines that are parsing dither patterns in different ways. Furthermore we need the information that you figured out here for places where parse_dither_pattern is used, i.e. for quicklook and I presume also for 2d coadding where we need the offsets. It would take someone a lot of time to figure that out in the future, whereas now you have the mosfire header cards and dither patterns at your fingertips.  

This does not look like a very difficult change since I think all the information is in the fitstable. I realize a few other instruments are involved, but that is part of how we code pypeit, i.e. to not just solve one instruments but all instruments.

Reply:
@jhennawi you are not looking at the latest changes. I do clearly state that this method, specifically for MOSFIRE, parse the dither pattern.

I think you got a different understanding of the philosophy of this method. When I wrote it, I did not mean it to be a dither pattern parsing, but it's meant to change the `comb_id` and `bkg_id` (how it's stated in the first sentence of the docs), since those values are assigned as default in `metadata.set_combination_groups`. In a generic usage, this method can assign  `comb_id` and/or `bkg_id` following the dither pattern or not and adapt to any other logic that the instrument requires. 
This is the reason why I don't want to replace `parse_dither_pattern` with this method. I could see using `parse_dither_pattern`  inside this method, but I will have to make substantial changes to  `parse_dither_pattern` and all the quicklook scripts that use it, and this is beyond the scope of this PR.  We can add a TODO for the future.

Reply:
Hi @deborape perhaps this is an issue with GitHub, but I'm looking here: 

https://github.com/pypeit/PypeIt/blob/bfb3b1b6475d37ee744d3e1169f4b29bee57e21d/pypeit/spectrographs/keck_mosfire.py

And I see no mention of parsing dither patterns in the documentation for get_comb_group.  This needs to be clearly stated in the get_comb_group Docstring as well as in the dummy method in the spectrograph.py. 

As for the parse_dither_pattern issue, I'm fine with making that a to do for the future.

Reply:
Okay thanks, it is strange that GitHub was not updated so I could see your docs. Sorry about that!